_posts/2021-05-30-spring-elasticache-redis.md

```diff
@@ -0,0 +1,235 @@
+---
+title: "Caching with ElastiCache for Redis and Spring Cloud AWS"
+categories: [craft]
+date: 2021-05-31 00:00:00 +1100
+modified: 2021-05-31 00:00:00 +1100
+author: mmr
+excerpt: "In this article, we will look at configuring a Spring Boot Application to use AWS ElastiCache for Redis using Spring Cloud AWS"
+image:
+ auto: 0090-404
+---
+
+ElastiCache is a fully managed caching service available in AWS Cloud. 
+
+[Spring Cloud AWS](https://docs.awspring.io/spring-cloud-aws/docs/current/reference/html/index.html) helps us to simplify the communication of Spring Boot application with AWS services. From taking care
+of security to auto-configuring the beans required for the communication, it takes care of a lot of essential steps.
+
+In this article, we will look at how we can use it to connect our application to [AWS ElastiCache for Redis](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/SelectEngine.html).
+ 
+{% include github-project.html url="https://github.com/thombergs/code-examples/tree/master/aws/spring-cloud-caching-redis" %}
+
+## Why Caching?
+
+Caching is a common technique of temporarily storing a copy of data or result of a computation in memory for quick and frequent access. We use caching primarily for :
+
+1. Improving the throughput of the application.
+2. Prevent overwhelming the application or downstream applications with redundant requests.
```

Comment:
This is actually an upstream service, see https://reflectoring.io/upstream-downstream/#upstream-and-downstream-micro-services. This is why I don't like using the terms upstream and downstream: they are confusing.

I changed the wording here.